src/engine/GroupBy.cpp

```diff
@@ -189,18 +192,22 @@ void processGroup(const GroupBy::Aggregate& a, size_t blockStart,
           }
         }
       } else if (inputTypes[a._inCol] == ResultTable::ResultType::FLOAT) {
+        // used to store the id value of the entry interpreted as a float
+        float tmpF;
         if (a._distinct) {
           for (size_t i = blockStart; i <= blockEnd; i++) {
             const auto it = distinctHashSet.find((*input)[i][a._inCol]);
             if (it == distinctHashSet.end()) {
               distinctHashSet.insert((*input)[i][a._inCol]);
-              res += *reinterpret_cast<const float*>(&(*input)[i][a._inCol]);
+              std::memcpy(&tmpF, &(*input)[i][a._inCol], sizeof(float));
```

Comment:
Do we need to first derefence and then take the address?

Reply:
input could be a pointer to a vector of vectors, or a pointer to a vector of arrays so the dereferencing is required to access the proper elment, of which we then need the address. Adding more parantheses could make the intended effect of the code clearer, might also make it harder to read though.